### PR TITLE
Fix marker offset not being updated after start button is pressed

### PR DESCRIPTION
**Pull request by raccube**

### CODE REVIEW

File: sr/robot3/robot.py [hunk 1]
```diff
@@ -168,6 +168,10 @@ def is_arduino(cls, port: ListPortInfo) -> bool:
     def _init_metadata(self) -> None:
         """Fetch metadata from Astoria."""
         self._metadata, self._code_path = GetMetadataConsumer.get_metadata()
+        if hasattr(self, '_cameras'):
```
Comment (trickeydan) [inline sr/robot3/robot.py:172]: Pretty sure this attribute always exists?

File: sr/robot3/robot.py [hunk 2]
```diff
@@ -168,6 +168,10 @@ def is_arduino(cls, port: ListPortInfo) -> bool:
...
+            for camera in self._cameras:
+                # type: ignore[attr-defined]
+                camera._backend._zcam._marker_offset = self._metadata.marker_offset
```
Comment (trickeydan) [inline sr/robot3/robot.py:171]: This might need a type: ignore

File: sr/robot3/robot.py (hunk 1 above)
Comment (PeterJCLaw) [reply sr/robot3/robot.py:172]: It gets created as part of the initialisation, currently the metadata init happens before the cameras init.

File: sr/robot3/robot.py (hunk 2 above)
Comment (raccube) [reply sr/robot3/robot.py:171]: It's there, I put it on the wrong line at first

**Review by trickeydan — APPROVED** [review]

**Pull request merged**